src/connector/src/sink/mod.rs

```diff
@@ -66,6 +82,29 @@ pub trait Sink {
     async fn abort(&mut self) -> Result<()>;
 }
 
+pub struct SinkWriterInfo {
+    // TODO: add fields
+}
+
+#[async_trait]
+pub trait SinkCoordinator {
+    async fn init(&mut self, writer_info: Vec<SinkWriterInfo>) -> Result<()>;
+    async fn commit(&mut self, epoch: u64, metadata: Vec<Bytes>) -> Result<()>;
+}
+
+pub struct NoSinkCoordinator;
```

Comment:
```suggestion
pub struct DummySinkCoordinator;
```

Let's make our naming less confusing 🤣

Reply:
I think the current naming is acceptable. 

The term `dummy` means that we able to use it, and calling any of its method is no-op. 

However, in this PR, since `Coordinator` is a associated type for the trait `Sink`, each implementation of the trait should specify its coordinator type, even though it does not have a coordinator, such as jdbc and kafka sink. Therefore for sink without a coordinator, their coordinator type is the `NoSinkCoordinator`, and calling any of its method will panic to warn that we are using it incorrectly.

Reply:
I am concerning using `DummySinkCoordinator`, it can simply return `OK(())`. Then we can unify the sink procedure on the arch level and we may use duck typing to add more connectors in the future, without paying attention to whether it needs coordinator or not.